test/cbmc/proofs/ingestDataBlock/ingestDataBlock_harness.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * AWS IoT Over-the-air Update v3.1.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file ingestDataBlock_harness.c
+ * @brief Implements the proof harness for ingestDataBlock function.
+ */
+/*  Ota Agent includes. */
+#include "ota.h"
+#include "stubs.h"
+#include <stdlib.h>
+
+extern OtaAgentContext_t otaAgent;
+extern IngestResult_t ingestDataBlock( OtaFileContext_t * pFileContext,
+                                       const uint8_t * pRawMsg,
+                                       uint32_t messageSize,
+                                       OtaPalStatus_t * pCloseResult );
+
+IngestResult_t ingestDataBlockCleanup( OtaFileContext_t * pFileContext,
+                                       OtaPalStatus_t * pCloseResult )
+{
+    IngestResult_t result;
+
+    __CPROVER_assume( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ) );
+
+    __CPROVER_assert( pFileContext != NULL,
+                      "Error: Expected a non-NUll value for pFileContext" );
+
+    __CPROVER_assert( pCloseResult != NULL,
+                      "Error: Expected a non-NUll value for pCloseResult" );
+
+    return result;
+}
+
+IngestResult_t decodeAndStoreDataBlock( OtaFileContext_t * pFileContext,
+                                        const uint8_t * pRawMsg,
+                                        uint32_t messageSize,
+                                        uint8_t ** pPayload,
+                                        uint32_t * pBlockSize,
+                                        uint32_t * pBlockIndex )
+{
+    IngestResult_t result;
+
+    __CPROVER_assume( ( result >= IngestResultUninitialized ) && ( result <= IngestResultDuplicate_Continue ) );
```

Comment:
Similarly here: Why is this assumption on the result required? I think `ingestDataBlock` can handle all possible values. Do correct me if I am wrong.